python/ray/train/v2/api/exceptions.py

```diff
@@ -0,0 +1,17 @@
+from typing import Dict
+
+from ray.train.v2._internal.exceptions import RayTrainError
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class TrainingFailedError(RayTrainError):
+    """Exception raised by `<Framework>Trainer.fit()` when training fails."""
+
+    def __init__(self, error_message: str, worker_failures: Dict[int, Exception]):
+        super().__init__("Training failed due to worker errors:\n" + error_message)
+        self._error_message = error_message
+        self.worker_failures = worker_failures
```

Comment:
Now that we expose the error message, do we still need to include `worker_failures`? Or keep in the error message instructions on how to inspect.

Reply:
I was thinking that pickling along the errors basically allows the `fail_fast='raise'` functionality to be possible in case users want to check the actual errors that happened on workers.

I also didn't want this to be a fully public API and only making it public after more user feedback that it would actually be helpful.

Reply:
I was also thinking that creating the error string from the worker failures might be a better way to do this, rather than pass the error string in separately.